src/Network/WebSockets/TSSocketManager.m

```diff
@@ -98,10 +114,12 @@ - (void)becomeActive
         switch ([socket readyState]) {
             case SR_OPEN:
                 DDLogVerbose(@"WebSocket already open on connection request");
+                OWSAssert(self.status == kSocketStatusOpen);
```

Comment:
I've added asserts around the key `status` property. The logic that updates the status here, for example, should be redundant but I've left it as it stands in order to be more redundant in production.